src/adagrams.js

```diff
@@ -1,15 +1,160 @@
+const pool = {
+  A: 9,
+  B: 2,
+  C: 2,
+  D: 4,
+  E: 12,
+  F: 2,
+  G: 3,
+  H: 2,
+  I: 9,
+  J: 1,
+  K: 1,
+  L: 4,
+  M: 2,
+  N: 6,
+  O: 8,
+  P: 2,
+  Q: 1,
+  R: 6,
+  S: 4,
+  T: 6,
+  U: 4,
+  V: 2,
+  W: 2,
+  X: 1,
+  Y: 2,
+  Z: 1,
+};
+
 export const drawLetters = () => {
-  // Implement this method for wave 1
+  const characters = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
+  let hand = [];
+  let handFreq = {};
+
+  while (hand.length < 10) {
+    let randomInt = Math.floor(Math.random() * 26);
+    let randomLetter = characters.charAt(randomInt);
+
+    if (!hand.includes(randomLetter)) {
+      hand.push(randomLetter);
+      handFreq[randomLetter] = 1;
+    } else {
+      if (handFreq[randomLetter] < pool[randomLetter]) {
+        hand.push(randomLetter);
+        handFreq[randomLetter] += 1;
+      }
+    }
+  }
+
+  return hand;
 };
 
 export const usesAvailableLetters = (input, lettersInHand) => {
-  // Implement this method for wave 2
+  for (let char of input) {
+    if (lettersInHand.includes(char)) {
+      const index = lettersInHand.indexOf(char);
+      lettersInHand.splice(index, 1);
```

Comment:
We should not be removing the letters from `lettersInHand`. This function should only be checking to see if all of the letters in `input` are in `lettersInHand`. If we're removing letters from the hand each time we're doing this check, the user is going to run out of letters quick.